### ISSUE TITLE
Feature Request: Warnings for missing Aria properties in debug mode

### ISSUE THREAD

**Issue by tracker1** [facebook/react#15649]
Currently there are several default behaviors that should have associated aria attributes, for example alt text for `<img>` and `<svg>` or components with certain navigation or mouse event behaviors.

It would be really cool if those conditions that would be pointed out by tools like Lighthouse/WAVE/axe could issue `console.warn` statements on render, where reasonable for non-production builds.  This could guide developers towards at least taking care of a lot of the low hanging fruit in this regard.

Possibly done in conjunction with an `ariaset` property as mentioned in #1259.

**Comment by nmain** [comment]
Isn't this well covered by https://www.npmjs.com/package/eslint-plugin-jsx-a11y already?

**Comment by oliviertassinari** [comment]
Will we have a new tool after https://twitter.com/jwyung/status/1128342470356422656?

**Comment by tracker1** [comment]
@nmain 
> Isn't this well covered by https://www.npmjs.com/package/eslint-plugin-jsx-a11y already?

Does that cover external components that won't have the properties on the final elements if you don't pass them in?  Again, mainly thinking of at least a warning in non-production builds in the library.

**Comment by robertknight** [comment]
ESLint plugins only work on the code that you run it on, which typically excludes anything in `node_modules/`. Have you looked into [react-axe](https://github.com/dequelabs/react-axe)?

**Comment by threepointone** [comment]
I think react-axe does what you want? I'll close this issue since a good workaround exists, feel free to keep discussing.
